docs/writing/posts/retrieval-is-more-than-pure-chance.md

```diff
@@ -0,0 +1,675 @@
+
+We can do this by taking advantage of the `to_batches()` method that the `LanceTable` provides. This returns a `pyArrow` dataset that provides a`RecordBatch` object which can be consumed as an iterator.
+
+```python
+def fetch_chunks(table: LanceTable, n=int) -> Iterable[TextChunk]:
+    for batch in table.to_lance().to_batches():
+        chunk_ids = batch["chunk_id"]
+        text_arr = batch["text"]
+        for id, text in zip(chunk_ids, text_arr):
+            yield TextChunk(chunk_id=str(id), text=str(text))
+```
+
+??? info "What's in a batch?"
+    The `RecordBatch` object is a `PyArrow` specific object and returns the specific properties of each row. We therefore need to zip the chunk_ids and the texts that we get from the batch together to match the chunk_id to its specific text paragraph.
+
+    Once we do so, we convert the `StringScalar` object into its string representation so that we can use it in our query method
+
+This in turn gives us a generator containing `TextChunk` objects that we can consume like any other normal list. Once we've fetched the relevant chunks, we need to generate the relevant synthethic questions. We can use our `generate_questions` that we defined earlier to batch the generation of these questions easily with a small modification.
+
+This time, we'll set it to accept the `TextChunk` object instead so that we always get the generated question with the original chunk information. This helps prevent any potential errors arising from mixing up the generated question with the wrong chunk.
+
+```python
+async def generate_question_answer_pair(
+    chunk: TextChunk,
+) -> tuple[QuestionAnswerPair, TextChunk]:
+    return (
+        await client.chat.completions.create(
+            model="gpt-3.5-turbo",
+            messages=[
+                {
+                    "role": "system",
+                    "content": "You are a world class algorithm that excels at generating great questions that can be only answered by a specific text that will soon be passed to you. ",
+                },
+                {
+                    "role": "assistant",
+                    "content": f"Generate a question and answer pair that uses information and content that is specific to the following text chunk, including a chain of thought:\n\n{chunk}",
+                },
+            ],
+            response_model=QuestionAnswerPair,
+        ),
+        chunk,
+    )
+
+async def generate_questions(
+    chunks: Iterable[TextChunk],
+) -> List[tuple[QuestionAnswerPair, TextChunk]]:
+    coros = [generate_question_answer_pair(chunk) for chunk in chunks]
+    return await asyncio.gather(*coros)
+```
+
+We can then just recycle our original `get_response` methods that we used to query our database along with the evaluation metrics we defined as seen in the code below.
+
+```python
+db = connect("./db")
```

Comment:
you're showing too many details, this is useful for a tutorial but not for a blog post